src/main/java/com/ericsson/ei/rules/RulesHandler.java

```diff
@@ -45,32 +40,31 @@
 import com.fasterxml.jackson.databind.ObjectMapper;
 
 @Component
-@Scope(value = "thread", proxyMode = ScopedProxyMode.TARGET_CLASS)
+@Scope("thread")
 public class RulesHandler {
     private static final Logger LOGGER = LoggerFactory.getLogger(RulesHandler.class);
 
-    private static final String EI_HOME_DEFAULT_NAME = ".eiffel-intelligence";
-
     @Value("${rules.path}")
     private String rulesFilePath;
 
     private JmesPathInterface jmesPathInterface = new JmesPathInterface();
     private static String jsonFileContent;
     private static JsonNode parsedJson;
 
-    public RulesHandler() {
+    public RulesHandler() throws Exception {
         super();
     }
 
     @PostConstruct
-    public void init() {
+    public void init() throws Exception {
         if (parsedJson == null) {
             try {
-                jsonFileContent = readRulesFileContent(rulesFilePath);
+                jsonFileContent = readRulesFileContent();
                 ObjectMapper objectmapper = new ObjectMapper();
                 parsedJson = objectmapper.readTree(jsonFileContent);
             } catch (Exception e) {
-                LOGGER.error("RulesHandler Init: Failed to read Rules file: " + rulesFilePath, e.getMessage(), e);
+                LOGGER.error("Rules file failed to be loaded/read. Path: {}", rulesFilePath);
+                throw e;
```

Comment:
Since you throw e here and catch it later in the code,,, move the log message to wherever the exception catched. Include exception in the log message, so the stacktrace and log message is not spread out in the log.

Reply:
It's not caught later, this is the last chance to catch it. RulesHandler is a component and will be initialized by Spring at start up and PostConstruct is run after creation of the instance. I just added a  log message for the sake of having some input in the log. The exception is thrown out as a means to stop the spring application.

Reply:
ok,,, but what you are doing now is that you decoupling the log message of the fault and the stack-trace the logs. I think exception should be included in the log message so it becomes easier to follow the fault in the logs.

Reply:
Then perhaps I shouldn't log anything at all then. It will look strange to double print the exception since right after printing it into the log it will be followed by another stack trace print which will close the app. I'm not sure what the others think of this.

Reply:
try run EI with 10k+++ events and try to debug it(a lot of threads is running and writing to the log). Stacktrace will for sure be decoupled with probably 100+ log messages between. Try to debug that.

Reply:
I see your point. Then I think I'll just remove the logger message and just throw the exception all the way out. It's not super useful message anyway.

Reply:
I just realized though that this a process that should only run once upon creation of the component so it wouldn't be possible to get 100s of these at the same time...